### ISSUE TITLE
Mark occurrences of variable under the cursor

### ISSUE THREAD

**Issue by HighCommander4** [RustDT/RustDT#128]
CDT has a nifty feature called "Mark occurrences", where, when the cursor is over the name of a variable, other uses of the same variable in the file are highlighted. This is a really useful aid when reading code.

It would be great if RustDT had a similar feature.

**Comment by Woyten** [comment]
I have an idea how that feature could be realized. The idea involves `racer` searching for every String occurrence of the variable name that maps to the same location. In order to keep the performance reasonable, the search should be performed within some boundaries which could be determined by `parse-describe`.
Alternatively, the canonical way would be to have a real semantic understanding of the code. But currently, I believe, the rust compiler is not yet in that state...

**Comment by bruno-medeiros** [comment]
There is already some work being done in rustw with regards to this feature: https://twitter.com/nick_r_cameron/status/735335631228968960
Once it matures, all that will be left will be to integrate it with IDEs.

**Comment by HighCommander4** [comment]
So in the future RustDT will use rustw?

**Comment by bruno-medeiros** [comment]
> So in the future RustDT will use rustw?

Seems likely, but it's just an idea at the moment.

**Comment by HighCommander4** [comment]
> > So in the future RustDT will use rustw?
> 
> Seems likely, but it's just an idea at the moment.

Another idea, which I've just discovered, might be RLS: https://github.com/jonathandturner/rustls

**Comment by bruno-medeiros** [comment]
Yeah, I'm already working on LSP integration, which in the case of RustDT will eventually cover rustls
